mypy/build.py

```diff
@@ -833,6 +836,11 @@ def compute_hash(text: str) -> str:
     return hashlib.md5(text.encode('utf-8')).hexdigest()
 
 
+def compute_module_hash(path: str) -> str:
+    with open(path, 'r') as f:
+        return compute_hash(f.read())
```

Comment:
Then perhaps we should hash the binary, and not even bother with decoding to a string in case when we need to read the file from disk?

And yes, sometimes `.source` is already populated; but it's populated with a string. So I'm plannnig to split `read_with_python_encoding` into reading the binary, and then decoding, so that in between the two steps we can calculate and store the hash.

If that's too troublesome, I can instead standardize the hash to always use the string.

Any preferences?